src/foundations/layout/Grid.props.ts

```diff
@@ -25,6 +27,10 @@ type PGrid = {
     padding?: TSpacingDefinition;
     /** margin according to `TSpacingDefinition` type */
     margin?: TSpacingDefinition;
+    /** restrict the width of a `Grid` */
+    width?: number | null;
+    /** restrict the height of a `Grid` */
+    height?: number | null;
     children?: React.ReactNode;
```

Comment:
I am still 3/5 on positioning the Shape component relatively when we render child components inside it and not having to rely on rendering another Grid component inside.

Reply:
and you are right ... we currently do that
This is for another use-case. :D